### PR TITLE
Fix for argmax/argmin with cpptile2dkernel

### LINKED ISSUE #178964: [inductor] `max` returns incorrect indices when compiling a function includes `(a + b).max(dim=1)`
### 🐛 Describe the bug

The following operator sequence trigger this inconsistency between torch.compile and eager: `(a + b).max(dim=1)`, where `a` is a transposed tensor. Specifically, the returned indices of `max` operation are all zeros instead of correct indices.

Here is the code to reproduce:
```python
import torch
def fn(a, b):
    return (a + b).max(dim=1)

torch.manual_seed(0)
a = torch.randn(8, 8).transpose(0, 1)  # the shape should be (8, 8) to trigger the error
b = torch.randn(8, 8)

res1 = fn(a, b)
print(res1.indices)  # tensor([5, 5, 6, 7, 0, 3, 7, 2])
res2 = torch.compile(fn, backend='inductor')(a, b)
print(res2.indices)  # tensor([0, 0, 0, 0, 0, 0, 0, 0])
torch.testing.assert_close(res1, res2, equal_nan=True)  # assertion error

```

Interestingly, if we execute `a+b` outside of the compiled function, no error occurs:

```python

# This works fine
def fn2(x):
    return x.max(dim=1)
res1 = fn2(a+b)
print(res1.indices)  # tensor([5, 5, 6, 7, 0, 3, 7, 2])
res2 = torch.compile(fn2)(a+b)
print(res2.indices)  # tensor([5, 5, 6, 7, 0, 3, 7, 2])
torch.testing.assert_close(res1, res2, equal_nan=True)  # works fine
```

This issue happens when using backend `inductor`, and does not happen when using backend `aot_eager` or `eager`.

### Error logs

_No response_

### Versions

[pip3] numpy==2.4.2
[pip3] torch==2.12.0.dev20260328+cpu
[pip3] torchvision==0.26.0.dev20260223+cpu
[pip3] triton==3.6.0
[conda] numpy 2.4.2 pypi_0 pypi
[conda] torch 2.12.0.dev20260328+cpu pypi_0 pypi
[conda] torchvision 0.26.0.dev20260223+cpu pypi_0 pypi
[conda] triton 3.6.0 pypi_0 pypi

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by trichmo**
Fixes #178964

The root cause is that for CppTile2DKernel, the index used in the argmax and argmin calculations are incorrect. The value calculation is correctly determined using the linearized index. The reduce index was incorrectly missing the offset within the tile. The fix updates the reduce index to properly capture the offset.

Without transpose, the stride pattern of the two matrices match and CppVecKernel can be called which is why this was not hit without transpose. Similarly, the issue only would arise for >=8 dimension size because that is when vectorization begins to be utilized as determined in select_tiling. 

Before changes tmp_acc1_vec uses x1 (always) for storing the argmax value. _Looking at the penultimate line of code_
Screenshot From 2026-04-06 19-45-42

After changes tmp_acc1_vec uses x2 + x2_inner to properly store index results
Screenshot From 2026-04-06 19-48-26

Testing over argmax and argmin with both 3d and 2d matrix inputs, number of tiles, and the presence of a tail.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude are those CI failures related?

**Review by jansel — APPROVED** [review]
